src/library/scala/collection/immutable/TreeMap.scala

```diff
@@ -48,12 +48,13 @@ object TreeMap extends ImmutableSortedMapFactory[TreeMap] {
  *  @define mayNotTerminateInf
  *  @define willNotTerminateInf
  */
-final class TreeMap[A, +B] private (tree: RB.Tree[A, B])(implicit val ordering: Ordering[A])
+final class TreeMap[A, +B] private (_tree: RB.Tree[A, B])(implicit val ordering: Ordering[A])
   extends SortedMap[A, B]
      with SortedMapLike[A, B, TreeMap[A, B]]
      with MapLike[A, B, TreeMap[A, B]]
      with Serializable
      with HasForeachEntry[A, B] {
+  private[immutable] val tree: RB.Tree[A, B] = _tree
```

Comment:
Is this different from making it a val directly in the parameter list?

Reply:
The has changed since you commented to `private def tree0: RB.Tree[A, B] = tree`.

The problem we have is that `class C(x: A)` and `class C(private val x: A); object C { (_:C).x }` are not serialization compatible because the reference from `object C` caused the compiler to emit the backing field for that val with a mangled name.